backend/src/main/java/atlantbh/restaurants/services/UserService.java

```diff
@@ -0,0 +1,46 @@
+package atlantbh.restaurants.services;
+
+import atlantbh.restaurants.controllers.dto.UserRegistrationDTO;
+import atlantbh.restaurants.models.Location;
+import atlantbh.restaurants.models.User;
+import atlantbh.restaurants.repositories.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.web.bind.annotation.RequestBody;
+
+import javax.swing.text.html.Option;
+import javax.validation.Valid;
+import java.util.Optional;
+
+@Service
+public class UserService extends BaseService<User, UserRepository> {
+
+    @Autowired
+    LocationService locationService;
+
+    public User findByEmail(String email) {
+        return repo.findByEmail(email);
+    }
+
+    public User checkLoginData(String email, String password){
+        String passwordHash = hashPassword(password);
+        return repo.findByEmailAndPasswordHash(email, passwordHash);
+    }
+
+    public Boolean isEmailTaken(String email) {
+        return repo.existsUserByEmail(email);
+    }
+    public User register(@Valid @RequestBody UserRegistrationDTO user) {
+        String passwordHash = hashPassword(user.getPassword());
+        // Registration of new users is only possible for normal users
+        Optional<Location> location = locationService.getById(user.getLocationId());
```

Comment:
this method should be just called .get (id is default property)
Also, what if there is no location? You should always check if optional is present, before you call optional.get()

Reply:
Fixed in 92c4f72

Reply:
not good enough.. 
```
Location location = locationService.get(user.getLocationId()); 
// Since get method will throw ServiceException you don't need any special error handling, just throw it here to controller..
```

Reply:
Fixed in c2ed9b4060983e1312e68ee9cdd1de2be1986207